src/database/migrations/20231220235718_RemoveParticipantIdColumnFromAuditTrail.ts

```diff
@@ -0,0 +1,39 @@
+import { Knex } from 'knex';
+
+export async function up(knex: Knex): Promise<void> {
+  await knex('auditTrails')
+    .whereNotNull('participantId')
+    .update({
+      eventData: knex.raw("JSON_MODIFY(eventData, '$.participantId', ?)", [
+        knex.ref('participantId'),
+      ]),
+    });
+
+  await knex.schema.alterTable('auditTrails', (table) => {
+    table.dropForeign('participantId');
+    table.dropColumn('participantId');
+  });
+
+  await knex.schema.alterTable('businessContacts', (table) => {
+    table.dropForeign('participantId');
+    table.foreign('participantId').references('participants.id').onDelete('CASCADE');
+  });
+}
+
+export async function down(knex: Knex): Promise<void> {
+  await knex.schema.alterTable('auditTrails', (table) => {
+    table.integer('participantId').references('participants.id');
+  });
+
+  await knex('auditTrails')
+    .where(knex.raw("JSON_VALUE(eventData, '$.participantId') IS NOT NULL"))
+    .update({
+      participantId: knex.raw("JSON_VALUE(eventData, '$.participantId')"),
+      eventData: knex.raw("JSON_MODIFY(eventData, '$.participantId', null)"),
+    });
+
+  await knex.schema.alterTable('businessContacts', (table) => {
```

Comment:
Shouldn't `down` be re-instating the FK?

Reply:
I'm not totally sure what you mean sorry. I think it is?

Reply:
oh sorry got confused - I see that the cascade deletion is the thing that is changing.